mathjax3-ts/output/chtml/Wrappers/mtable.ts

```diff
@@ -0,0 +1,387 @@
+     */
+    protected handleFrame(frame: boolean) {
+        if (frame) {
+            (this.chtml.firstChild as HTMLElement).style.border = '.07em ' + this.node.attributes.get('frame');
+        }
+    }
+
+    /******************************************************************/
+
+    /*
+     * @param{string} name           The name of the attribute to get as an array
+     * @param{CHTMLWrapper} wrapper  The wrapper whose attribute is to be used
+     * @return{string[]}             The array of values in the given attribute, split at spaces,
+     *                                 padded to the number of table columns by repeating the last entry
+     */
+    protected getColumnAttributes(name: string, wrapper: CHTMLWrapper = null) {
+        const columns = this.getAttributeArray(name, wrapper);
+        if (columns.length === 0) return;
+        while (columns.length < this.numCols) {
+            columns.push(columns[columns.length - 1]);
+        }
+        return columns;
+    }
+
+    /*
+     * @param{string} name           The name of the attribute to get as an array
+     * @param{CHTMLWrapper} wrapper  The wrapper whose attribute is to be used
+     * @return{string[]}             The array of values in the given attribute, split at spaces,
+     *                                 padded to the number of table rows by repeating the last entry
+     */
+    protected getRowAttributes(name: string, wrapper: CHTMLWrapper = null) {
+        const rows = this.getAttributeArray(name, wrapper);
+        if (rows.length === 0) return;
+        while (rows.length < this.numRows) {
+            rows.push(rows[rows.length - 1]);
+        }
+        return rows;
+    }
+
+    /*
+     * @param{string} name           The name of the attribute to get as an array
+     * @param{CHTMLWrapper} wrapper  The wrapper whose attribute is to be used
+     * @return{string[]}             The array of values in the given attribute, split at spaces
+     *                                 (after leading and trailing spaces are removed, and multiple
+     *                                  spaces have been collapsed to one).
+     */
+    protected getAttributeArray(name: string, wrapper: CHTMLWrapper = null) {
+        const value = ((wrapper || this).node.attributes.get(name) as string);
+        if (!value) return [];
+        return value.replace(/^\s+/, '').replace(/\s+$/, '').replace(/\s+/g, ' ').split(/ /);
+    }
+
+    /*
+     * Converts an array of dimensions (with arbitrary units) to an array of dimensions
+     *   in units of em's, dividing the dimension by n (defaults to 1).
+     *
+     * @param{string[]} list   The array of dimensions to be turned into em's
+     * @param{nunber} n        The number to divide each dimension by after converted
+     * @return{string[]}       The array of values converted to em's
+     */
+    protected convertLengths(list: string[], n: number = 1) {
+        if (!list) return;
```

Comment:
Can that be null?

Reply:
Yes.  It is called on the results of `getRowAttributes()` and `getColumnAttriubtes()`, as well as others, and these can return null values.